ql/src/test/results/clientpositive/llap/cbo_case_when_wrong_type.q.out

```diff
@@ -56,7 +56,7 @@ POSTHOOK: Input: default@t
 #### A masked pattern was here ####
 CBO PLAN:
 HiveProject($f0=[1])
-  HiveFilter(condition=[IN($0, 1:SMALLINT, 2:SMALLINT)])
+  HiveFilter(condition=[AND(CASE(=(CAST($0):INTEGER, 1), true, =(CAST($0):INTEGER, 2), true, false), IN($0, 1:SMALLINT, 2:SMALLINT, 3:SMALLINT))])
```

Comment:
This seems like a regression from HIVE-25734. Moreover the CBO plan here is different from the CBO plan of the equivalent query with `case when a = 1 ...` showing that there is some inconsistency between equivalent expressions.